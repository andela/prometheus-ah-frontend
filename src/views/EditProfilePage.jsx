import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import editProfileAction from '../actions/profile/editProfile.action';

import EditProfileForm from '../components/EditProfileForm';

/**
 * @EditProfilePage page
 */
class EditProfilePage extends Component {
  /**
   * render
   *
   * @return {void}
   */
  render() {
    const {
      location: { state: { profile = {} } = {} } = {},
      editProfile,
      history,
      user: { isAuthenticated = {} }
    } = this.props;

    if (!isAuthenticated) {
      history.push('/');
    }

    return (
      <div>
        <br />
        <div className="row">
          <div className="col-sm-3" />
          <div className="container col-sm-6 profile-padding">
            <EditProfileForm
              profile={profile}
              editProfile={editProfile}
              history={history}
            />
          </div>
          <div className="col-sm-3" />
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.auth,
});

const mapDispatchToProps = dispatch => ({
  editProfile: user => dispatch(editProfileAction(user)),
});

EditProfilePage.propTypes = {
  editProfile: PropTypes.func.isRequired,
  location: PropTypes.shape({
  }).isRequired,
  history: PropTypes.shape({
  }).isRequired,
  user: PropTypes.shape({
  }).isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(EditProfilePage);
