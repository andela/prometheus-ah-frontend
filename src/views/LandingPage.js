import React from "react";
import { Link } from "react-router-dom";
import routes from "../constants/routes";

const LandingPage = () => (
  <div>
    This is from the LandingPage component
    <Link to={routes.SIGN_IN}>SIGN_IN</Link>
  </div>
);

export default LandingPage;
