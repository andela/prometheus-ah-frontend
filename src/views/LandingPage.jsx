import React, { Component } from 'react';
import data from './articlesData';
import Footer from '../components/Footer';
import RatedArticles from '../components/landingPage/ratedArticles/RatedArticles';
import FeaturedArticles from '../components/landingPage/featuredArticles/FeaturedArticles';
import RecentPost from '../components/landingPage/recentPosts/RecentPosts';
import FeaturedAuthor from '../components/landingPage/featuredAuthor/FeaturedAuthor';

/**
 * @class LandingPage
 */
class LandingPage extends Component {
  /**
   * @description Render the JSX template
   *
   * @memberof LoginModal
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    const ratedPosts = data.articles.slice(0, 5);
    const featuredPosts = data.articles.slice(0, 5);
    const recentPosts = data.articles.slice(-3);
    const authorPosts = data.articles.slice(4, 7);
    const featuredAuthor = data.user;

    return (
      <React.Fragment>
        <RatedArticles posts={ratedPosts} />
        <section className="main-content my-5">
          <div className="row">
            <FeaturedArticles posts={featuredPosts} />
            <div className="col-12 col-md-5 col-lg-5">
              <FeaturedAuthor posts={authorPosts} author={featuredAuthor} />
              <RecentPost posts={recentPosts} />
            </div>
          </div>
        </section>
        <Footer />
      </React.Fragment>
    );
  }
}

export default LandingPage;
