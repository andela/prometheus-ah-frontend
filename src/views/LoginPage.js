import React from 'react';
import Header from '../components/Header';

const LoginPage = () => (
  <div>
    <Header />
    <div>
      This is from the Login component
    </div>
  </div>
);

export default LoginPage;
