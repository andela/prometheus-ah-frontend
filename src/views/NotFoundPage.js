import React from 'react';
import { Link } from 'react-router-dom';
import Footer from '../components/Footer';

const NotFoundPage = () => (
  <div>
    <div className="not-found">
      <h1>Sorry, we are yet to write that article</h1>
      <p>Please click the button to check out our other articles</p>
      {' '}
      <Link to="/"><button type="button" className="not-found-btn">Go home</button></Link>
    </div>
    <Footer />
  </div>
);

export default NotFoundPage;
