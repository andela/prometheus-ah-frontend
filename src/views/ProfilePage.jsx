import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import routes from '../constants/routes';
import userProfile from '../actions/profile/userProfile.action';
import userArticles from '../actions/profile/userArticles.action';
import userFollowers from '../actions/profile/userFollowers.action';
import userFollowing from '../actions/profile/userFollowing.action';
import Profile from '../components/Profile';
import Loading from '../components/common/Loading';

/**
 * Profile page
 */
class ProfilePage extends Component {
  /**
   * componentDidMount
   *
   * @return {void}
   */
  componentDidMount() {
    const {
      userProfileAction, userArticlesAction, userFollowersAction, userFollowingAction
    } = this.props;
    const { user: { user: { username = {} } } } = this.props;
    Promise.all([
      userProfileAction(username),
      userArticlesAction(username),
      userFollowersAction(username),
      userFollowingAction(username)
    ]);
  }

  /**
   * render
   *
   * @return {void}
   */
  render() {
    const { loading } = this.props;
    if (loading) {
      return (<Loading />);
    }

    const {
      profile: { user: { profile = {} } = {} },
      articles: { user: { articles = {} } = {} },
      followers,
      following,
      user: { user: { userId } = {} },
      history,
      user: { isAuthenticated = {} }
    } = this.props;

    if (!isAuthenticated) {
      history.push('/');
    }


    if (Object.keys(profile).length === 0) {
      return (
        <div>
          <Link to={routes.LANDING}>Home Page</Link>
        </div>
      );
    }

    return (
      <div>
        <Profile
          profile={profile}
          articles={articles}
          followers={followers}
          following={following}
          userId={userId}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  loading: state.userProfile.loading,
  profile: state.userProfile,
  articles: state.userArticles,
  bookmarks: state.userBookmarks,
  followers: state.userFollowers,
  following: state.userFollowing,
  user: state.auth,
});

const mapDispatchToProps = dispatch => ({
  userProfileAction: user => dispatch(userProfile(user)),
  userArticlesAction: username => dispatch(userArticles(username)),
  userFollowersAction: username => dispatch(userFollowers(username)),
  userFollowingAction: username => dispatch(userFollowing(username)),
});

ProfilePage.propTypes = {
  userProfileAction: PropTypes.func.isRequired,
  userFollowersAction: PropTypes.func.isRequired,
  userFollowingAction: PropTypes.func.isRequired,
  userArticlesAction: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  profile: PropTypes.shape({
  }).isRequired,
  followers: PropTypes.shape({
  }).isRequired,
  following: PropTypes.shape({
  }).isRequired,
  user: PropTypes.shape({
  }).isRequired,
  articles: PropTypes.shape({
  }).isRequired,
  history: PropTypes.shape({
  }),
};

export default connect(mapStateToProps, mapDispatchToProps)(ProfilePage);
