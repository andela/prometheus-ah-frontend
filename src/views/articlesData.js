/* eslint-disable */
const data = {
  articles: [
    {
      "id": 1,
      "slug": "this-is-how-i-feel",
      "image": "https://image.ibb.co/eLVQwL/kids.jpg",
      "title": "My Broken Mother, My Broken Heart",
      "body": "My task is getting bigger and I am equal to the task. Music gives a soul to the universe, wings to the mind, flight to the imagination and life to everything. This doesn't make sense actually. I Just want to make long text",
      "userId": 1,
      "description": "feeling very excited",
      "readingTime": "1 min read",
      "createdAt": "2018-09-20T13:43:12.129Z",
      "updatedAt": "2018-09-20T13:48:33.958Z",
      "User": {
        "id": 1,
        "username": "Joeeasy",
        "email": "joeeasy@gmail.com",
        "firstname": "Tega",
        "lastname": "Okeremeta",
        "role": "user",
        "status": "active",
        "image": null
      },
      "Tags": ["Technolgy"]
    },
    {
      "id": 2,
      "slug": "this-is-how-i-feel-1",
      "image": "https://image.ibb.co/eLVQwL/kids.jpg",
      "title": "It’s Not What You Know, It’s How You Think",
      "body": "My task is getting bigger and I am equal to the task. Music gives a soul to the universe, wings to the mind, flight to the imagination and life to everything. This doesn't make sense actually. I Just want to make long text",
      "userId": 1,
      "description": "feeling very excited",
      "readingTime": "5 min read",
      "createdAt": "2018-09-20T13:49:49.763Z",
      "updatedAt": "2018-09-20T13:49:49.763Z",
      "User": {
        "id": 1,
        "username": "Faksam",
        "email": "joeeasy@gmail.com",
        "firstname": "Tega",
        "lastname": "Okeremeta",
        "role": "user",
        "status": "active",
        "image": null
      },
      "Tags": ["Technolgy"]
    },
    {
      "id": 3,
      "slug": "this-is-how-i-feel-2",
      "image": "https://image.ibb.co/eLVQwL/kids.jpg",
      "title": "This is how I feel",
      "body": "My task is getting bigger and I am equal to the task. Music gives a soul to the universe, wings to the mind, flight to the imagination and life to everything. This doesn't make sense actually. I Just want to make long text",
      "userId": 1,
      "description": "feeling very excited",
      "readingTime": "6 min read",
      "createdAt": "2018-09-20T14:06:09.630Z",
      "updatedAt": "2018-09-20T14:06:09.630Z",
      "User": {
        "id": 1,
        "username": "Dinma",
        "email": "joeeasy@gmail.com",
        "firstname": "Tega",
        "lastname": "Okeremeta",
        "role": "user",
        "status": "active",
        "image": null
      },
      "Tags": ["Technolgy"]
    },
    {
      "id": 4,
      "slug": "this-is-how-i-feel-2",
      "image": "https://image.ibb.co/eLVQwL/kids.jpg",
      "title": "How to Make Time to Make Things",
      "body": "My task is getting bigger and I am equal to the task",
      "userId": 1,
      "description": "feeling very excited",
      "readingTime": "6 min read",
      "createdAt": "2018-09-20T14:06:09.630Z",
      "updatedAt": "2018-09-20T14:06:09.630Z",
      "User": {
        "id": 1,
        "username": "Blaze",
        "email": "joeeasy@gmail.com",
        "firstname": "Tega",
        "lastname": "Okeremeta",
        "role": "user",
        "status": "active",
        "image": null
      },
      "Tags": ["Technolgy"]
    },
    {
      "id": 5,
      "slug": "this-is-how-i-feel-2",
      "image": "https://image.ibb.co/eLVQwL/kids.jpg",
      "title": "The End of Snap and Telsa",
      "body": "My task is getting bigger and I am equal to the task. Music gives a soul to the universe, wings to the mind, flight to the imagination and life to everything. This doesn't make sense actually. I Just want to make long text",
      "userId": 1,
      "description": "feeling very excited",
      "readingTime": "6 min read",
      "createdAt": "2018-09-20T14:06:09.630Z",
      "updatedAt": "2018-09-20T14:06:09.630Z",
      "User": {
        "id": 1,
        "username": "JustTega",
        "email": "joeeasy@gmail.com",
        "firstname": "Tega",
        "lastname": "Okeremeta",
        "role": "user",
        "status": "active",
        "image": null
      },
      "Tags": ["Technolgy"]
    },
    {
      "id": 6,
      "slug": "this-is-how-i-feel-2",
      "image": "https://image.ibb.co/eLVQwL/kids.jpg",
      "title": "This is how I feel",
      "body": "My task is getting bigger and I am equal to the task",
      "userId": 1,
      "description": "feeling very excited",
      "readingTime": "6 min read",
      "createdAt": "2018-10-17T08:06:09.630Z",
      "updatedAt": "2018-09-20T14:06:09.630Z",
      "User": {
        "id": 1,
        "username": "joeeasy",
        "email": "joeeasy@gmail.com",
        "firstname": "Tega",
        "lastname": "Okeremeta",
        "role": "user",
        "status": "active",
        "image": null
      },
      "Tags": ["Technolgy"]
    },
    {
      "id": 7,
      "slug": "this-is-how-i-feel-2",
      "image": "https://image.ibb.co/eLVQwL/kids.jpg",
      "title": "This is how I feel",
      "body": "My task is getting bigger and I am equal to the task",
      "userId": 1,
      "description": "feeling very excited",
      "readingTime": "6 min read",
      "createdAt": "2018-10-16T14:06:09.630Z",
      "updatedAt": "2018-09-20T14:06:09.630Z",
      "User": {
        "id": 1,
        "username": "joeeasy",
        "email": "joeeasy@gmail.com",
        "firstname": "Tega",
        "lastname": "Okeremeta",
        "role": "user",
        "status": "active",
        "image": null
      },
      "Tags": ["Technolgy"]
    },
    {
      "id": 8,
      "slug": "this-is-how-i-feel-2",
      "image": "https://image.ibb.co/eLVQwL/kids.jpg",
      "title": "This is how I feel",
      "body": "My task is getting bigger and I am equal to the task",
      "userId": 1,
      "description": "feeling very excited",
      "readingTime": "6 min read",
      "createdAt": "2018-09-20T14:06:09.630Z",
      "updatedAt": "2018-09-20T14:06:09.630Z",
      "User": {
        "id": 1,
        "username": "joeeasy",
        "email": "joeeasy@gmail.com",
        "firstname": "Tega",
        "lastname": "Okeremeta",
        "role": "user",
        "status": "active",
        "image": null
      },
      "Tags": ["Technolgy"]
    }
  ],
  user: {
    "id": 5,
    "username": "Blaze",
    "email": "joeeasy@gmail.com",
    "firstname": "Jon",
    "lastname": "Snow",
    "role": "user",
    "status": "active",
    "image": "https://image.ibb.co/chF3i0/author2.jpg"
  }
};

export default data;
