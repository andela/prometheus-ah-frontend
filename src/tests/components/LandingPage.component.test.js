import React from 'react';
import { shallow } from 'enzyme';
import { LandingPage } from '../../views/LandingPage';

let wrapper;

describe('LandingPage component', () => {
  beforeEach(() => {
    const props = {
      featuredArticles: [
        {
          id: 23,
          slug: 'nobody-taught-you',
          title: 'nobody taught you',
          body: 'Before dying at the age of 39, Blaise Pascal',
          userId: 2,
          description: 'all our problems',
          readingTime: '1 min read',
          createdAt: '2018-10-20T12:21:33.812Z',
          updatedAt: '2018-10-20T12:21:33.812Z',
          User: {
            username: 'faksam',
            firstname: null,
            lastname: null
          },
          Tags: [
            {
              name: 'Self'
            }
          ]
        },
      ],
      author: {
        featuredAuthor: {
          id: 2,
          username: 'faksam',
          firstname: null,
          lastname: null,
          image: null,
          articles: [
            {
              id: 23,
              slug: 'the-most-important-skill-nobody-taught-you',
              title: 'The most important skill nobody taught you',
              readingTime: '1 min read'
            }
          ]
        }
      },
      articles: [
        {
          id: 23,
          slug: 'nobody-taught-you',
          title: 'nobody taught you',
          body: 'Before dying at the age of 39, Blaise Pascal',
          userId: 2,
          description: 'all our problems',
          readingTime: '1 min read',
          createdAt: '2018-10-20T12:21:33.812Z',
          updatedAt: '2018-10-20T12:21:33.812Z',
          User: {
            username: 'faksam',
            firstname: null,
            lastname: null
          },
          Tags: [
            {
              name: 'Self'
            }
          ]
        }
      ],
      actions: {
        fetchFeaturedArticles: jest.fn(),
        fetchArticles: jest.fn(),
        fetchAuthor: jest.fn()
      }
    };

    wrapper = shallow(<LandingPage {...props} />);
  });

  test('should render a snapshot of the userNavigation component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
