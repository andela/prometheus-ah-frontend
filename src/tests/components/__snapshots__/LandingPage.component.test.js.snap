// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component should render a snapshot of the userNavigation component 1`] = `
<Fragment>
  <RatedArticles
    posts={
      Array [
        Object {
          "Tags": Array [
            "Technolgy",
          ],
          "User": Object {
            "email": "joeeasy@gmail.com",
            "firstname": "Tega",
            "id": 1,
            "image": null,
            "lastname": "Okeremeta",
            "role": "user",
            "status": "active",
            "username": "Joeeasy",
          },
          "body": "My task is getting bigger and I am equal to the task. Music gives a soul to the universe, wings to the mind, flight to the imagination and life to everything. This doesn't make sense actually. I Just want to make long text",
          "createdAt": "2018-09-20T13:43:12.129Z",
          "description": "feeling very excited",
          "id": 1,
          "image": "https://image.ibb.co/eLVQwL/kids.jpg",
          "readingTime": "1 min read",
          "slug": "this-is-how-i-feel",
          "title": "My Broken Mother, My Broken Heart",
          "updatedAt": "2018-09-20T13:48:33.958Z",
          "userId": 1,
        },
        Object {
          "Tags": Array [
            "Technolgy",
          ],
          "User": Object {
            "email": "joeeasy@gmail.com",
            "firstname": "Tega",
            "id": 1,
            "image": null,
            "lastname": "Okeremeta",
            "role": "user",
            "status": "active",
            "username": "Faksam",
          },
          "body": "My task is getting bigger and I am equal to the task. Music gives a soul to the universe, wings to the mind, flight to the imagination and life to everything. This doesn't make sense actually. I Just want to make long text",
          "createdAt": "2018-09-20T13:49:49.763Z",
          "description": "feeling very excited",
          "id": 2,
          "image": "https://image.ibb.co/eLVQwL/kids.jpg",
          "readingTime": "5 min read",
          "slug": "this-is-how-i-feel-1",
          "title": "It’s Not What You Know, It’s How You Think",
          "updatedAt": "2018-09-20T13:49:49.763Z",
          "userId": 1,
        },
        Object {
          "Tags": Array [
            "Technolgy",
          ],
          "User": Object {
            "email": "joeeasy@gmail.com",
            "firstname": "Tega",
            "id": 1,
            "image": null,
            "lastname": "Okeremeta",
            "role": "user",
            "status": "active",
            "username": "Dinma",
          },
          "body": "My task is getting bigger and I am equal to the task. Music gives a soul to the universe, wings to the mind, flight to the imagination and life to everything. This doesn't make sense actually. I Just want to make long text",
          "createdAt": "2018-09-20T14:06:09.630Z",
          "description": "feeling very excited",
          "id": 3,
          "image": "https://image.ibb.co/eLVQwL/kids.jpg",
          "readingTime": "6 min read",
          "slug": "this-is-how-i-feel-2",
          "title": "This is how I feel",
          "updatedAt": "2018-09-20T14:06:09.630Z",
          "userId": 1,
        },
        Object {
          "Tags": Array [
            "Technolgy",
          ],
          "User": Object {
            "email": "joeeasy@gmail.com",
            "firstname": "Tega",
            "id": 1,
            "image": null,
            "lastname": "Okeremeta",
            "role": "user",
            "status": "active",
            "username": "Blaze",
          },
          "body": "My task is getting bigger and I am equal to the task",
          "createdAt": "2018-09-20T14:06:09.630Z",
          "description": "feeling very excited",
          "id": 4,
          "image": "https://image.ibb.co/eLVQwL/kids.jpg",
          "readingTime": "6 min read",
          "slug": "this-is-how-i-feel-2",
          "title": "How to Make Time to Make Things",
          "updatedAt": "2018-09-20T14:06:09.630Z",
          "userId": 1,
        },
        Object {
          "Tags": Array [
            "Technolgy",
          ],
          "User": Object {
            "email": "joeeasy@gmail.com",
            "firstname": "Tega",
            "id": 1,
            "image": null,
            "lastname": "Okeremeta",
            "role": "user",
            "status": "active",
            "username": "JustTega",
          },
          "body": "My task is getting bigger and I am equal to the task. Music gives a soul to the universe, wings to the mind, flight to the imagination and life to everything. This doesn't make sense actually. I Just want to make long text",
          "createdAt": "2018-09-20T14:06:09.630Z",
          "description": "feeling very excited",
          "id": 5,
          "image": "https://image.ibb.co/eLVQwL/kids.jpg",
          "readingTime": "6 min read",
          "slug": "this-is-how-i-feel-2",
          "title": "The End of Snap and Telsa",
          "updatedAt": "2018-09-20T14:06:09.630Z",
          "userId": 1,
        },
      ]
    }
  />
  <section
    className="main-content my-5"
  >
    <div
      className="row"
    >
      <FeaturedArticles
        posts={
          Array [
            Object {
              "Tags": Array [
                Object {
                  "name": "Self",
                },
              ],
              "User": Object {
                "firstname": null,
                "lastname": null,
                "username": "faksam",
              },
              "body": "Before dying at the age of 39, Blaise Pascal",
              "createdAt": "2018-10-20T12:21:33.812Z",
              "description": "all our problems",
              "id": 23,
              "readingTime": "1 min read",
              "slug": "nobody-taught-you",
              "title": "nobody taught you",
              "updatedAt": "2018-10-20T12:21:33.812Z",
              "userId": 2,
            },
          ]
        }
      />
      <div
        className="col-12 col-md-5 col-lg-5"
      >
        <FeaturedAuthor
          author={
            Object {
              "featuredAuthor": Object {
                "articles": Array [
                  Object {
                    "id": 23,
                    "readingTime": "1 min read",
                    "slug": "the-most-important-skill-nobody-taught-you",
                    "title": "The most important skill nobody taught you",
                  },
                ],
                "firstname": null,
                "id": 2,
                "image": null,
                "lastname": null,
                "username": "faksam",
              },
            }
          }
        />
        <RecentPosts
          posts={
            Array [
              Object {
                "Tags": Array [
                  Object {
                    "name": "Self",
                  },
                ],
                "User": Object {
                  "firstname": null,
                  "lastname": null,
                  "username": "faksam",
                },
                "body": "Before dying at the age of 39, Blaise Pascal",
                "createdAt": "2018-10-20T12:21:33.812Z",
                "description": "all our problems",
                "id": 23,
                "readingTime": "1 min read",
                "slug": "nobody-taught-you",
                "title": "nobody taught you",
                "updatedAt": "2018-10-20T12:21:33.812Z",
                "userId": 2,
              },
            ]
          }
        />
      </div>
    </div>
  </section>
  <Footer />
</Fragment>
`;
